blockchain/lib/logic.js

```diff
@@ -61,6 +63,66 @@ async function shipmentTransfer(shipmentTransfer) {
     }
 }
 
+/**
+ * ShipmentTransfer transaction will handle three status : ASSIGN, CHECK_IN
+ * @param {org.bitship.ShipmentVehicleReport} shipmentVehicleReport
+ * @transaction
+ */
+async function shipmentVehicleReport(shipmentVehicleReport) {
+
+    const packageRegistry = await getAssetRegistry("org.bitship.Package");
+    let event = getFactory().newEvent('org.bitship', 'ShipmentTransfered');
+
+    // Update status for package
+    for (const packageObject of shipmentVehicleReport.packages) {
+        if (shipmentVehicleReport.status == "OK") {
+            packageObject.status = "IN_WAREHOUSE";
+            event.message = "Success";
+        } else {
+            packageObject.status = "LOST";
+            event.message = "Lost";
+        }
```

Comment:
We should compare list of packages to ShipmentVehicle.packages. If package does not contain in ShipmentVehicle.packages  => Change status to LOST